ospfd/ospf_vty.c

```diff
@@ -6077,6 +6077,11 @@ DEFUN (ip_ospf_cost,
 	char *coststr = NULL, *ifaddr = NULL;
 	coststr = argv_find(argv, argc, "(1-65535)", &idx) ? argv[idx]->arg
 							   : NULL;
+	if (coststr == NULL) {
+		vty_out(vty, "OSPF: Cost (1-65535) is not valid\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+
```

Comment:
this block isn't necessary as this condition will never be hit, lines 6077 - 6084 can be replaced with

```
argv_find(argv, argc, "(1-65535)", &idx);
coststr = argv[idx]->arg
```

Reply:
ok,